CONTRIBUTING.md

```diff
@@ -16,15 +16,186 @@ GitBook (which we use for documentation) uses Markdown as its file format.
+
+```
+---
+title: "New language support details"    # Title on page
+sidebarTitle: "New language"             # How title appears in sidebar
+weight: 6
+layout: single
+---
+```
+
+#### Disabling table of contents
+
+By default, all pages that use the `single` layout will generate a table of contents beneath their title. If you would like to disable it, add `toc` attribute to your front matter:
+
+```
+---
+title: "New language"
+weight: 4
+toc: false
+---
+```
+
+### Shortcodes
+
+There are a few shortcodes written for the documentation that keep styling consistent and insert specific HTML into pages where they are referenced.
+
+#### Notes
+
+For the moment, Hugo does not provide a way to pass values to Markdown's blockquote syntax, but in this documentation there are a few variations on the blockquote that are used for different purposes. So, when leaving a "Note" or blockquote on your pages, use the `note` shortcode instead
+
+```
+{{< note >}}
+Lorem ipsum
+{{< /note }}
+```
+
+This shortcode will automatically create the **Note:** first line for you.
+
+`note` accepts two optional parameters:
+
+- `theme`: default none, standard note. (`info`, `warning`). This will change the color and the default first line.
+- `title`: default dependent on theme. Over-rides that default.
+
+```
+{{< note theme="warning" title="Danger, Will Robinson" >}}
+Lorem ipsum!
+{{< /note >}}
+```
+
+#### Images
+
+The standard Markdown syntax for images remains the same, but it is possible to pass values to it that modify the appearance of the image
+
+```
+<!-- Normal image syntax -->
+![alt-title](/images/current_image.png)
+
+<!-- Scaled image -->
+![alt-title](/images/current_image.png "0.5")
+
+<!-- Scaled image displayed inline, such as an icon in a sentence -->
+![alt-title](/images/current_image.png "0.1-inline")
+```
+
+#### Videos & asciinema
+
+Displaying videos and asciinema recordings requires their own shortcodes, but they have identical parameters:
+
+```
+<!-- Video -->
+{{< video src="videos/management-console/new-project-creation.mp4" >}}
+
+<!-- Asciinema -->
+{{< asciinema src="videos/asciinema/verify-cli-extended.cast" >}}
+```
+
+#### Codetabs
+
+You can display tabbed content that will apply syntax highlighting to it where defined.
+
+```
+{{< codetabs >}}
+
+---
+title=Elasticsearch
+file=static/files/fetch/examples/php/elasticsearch
+highlight=php
+markdownify=false
+---
+
+<--->
+
+---
+title=Memcached
+file=none
+highlight=python
+markdownify=false
+---
+
+from jwcrypto import jws, jwk
+
+{{< /codetabs >}}
+```
+
+The first tab *Elasticsearch* will read from a local file and highlight it for PHP, whereas the second tab will apply Python highlighting to the content below `---`. Individual tabs are separated with the `<--->` line.
```

Comment:
What's markdownify all about, and what is the default?  (The default seems to be true, since everything I've seen specifies false, which feels backwards.)

Reply:
It will apply markdown by default to everything, except if the `file` you're using is a markdown file. Then you have to explicitly say make it markdown.